### ISSUE TITLE
When columns are thin and column spacing is set, items move from their grid position when clicked

### ISSUE THREAD

**Issue by natle01** [react-grid-layout/react-grid-layout#2014]
### Describe the bug

When a responsive layout has thin columns which are less than ~6% of the layout width, clicking on a layout item it will move it left (`x` position).
- `margin` column must be set.
- Thin columns can be created using any mixture of these properties `containerPadding`, `margin`, `width`, and `cols`.

### Your Example Website or App

https://codesandbox.io/p/sandbox/quirky-artem-tmpkws

### Steps to Reproduce the Bug or Issue

1. Set `width: 800` pixels.
1. Set `cols: {{ <yourbreakpoint>: 10 }}` 
Currently columns are 80px wide (10% of the layout width)
1. Set `margin: [40,0]`
Currently columns are 44px wide (5.5% of the layout width)

(Note. `margin` must be set)

### Expected behavior

When a responsive layout item is clicked, it should not move.
Layout item should only move when it is dragged to another `x` or `y` position.

### react-grid-layout library version

1.4.3

### Operating System Version

Windows 11

### Browser

Edge

### Additional context

Similar to this bug #1924 When containerPadding is 0 and cols is large enough, clicking on the event will trigger onLayoutChange, causing the layout to move left

### Screenshots or Videos

![MovingLayoutItem](https://github.com/react-grid-layout/react-grid-layout/assets/74999478/64375a7d-edf7-419f-a486-02acf977ada5)

**Comment by AndreiAyar** [comment]
Same here. Any solution ?

**Comment by xuanxiaoxiao** [comment]
mo too

**Comment by shamilium** [comment]
I have same issue.

**Comment by mvrozanti** [comment]
Yup, happens to me too. If I remove the margin parameter it doesn't move when I click it. But I still need the margin xd

**Comment by SLKarol** [comment]
I have same issue.

**Comment by STRML** [comment]
> *This was generated by AI during triage.*

— Items move from grid position on click

## Triage Notes

**What we've established so far:**

- The reported trigger (thin columns ≈ 5% width + margin set + click moves item by rounding) matches the rounding class that v2 addressed: `calcGridItemPosition` now corrects adjacent-item margins when idle (commit 618a763, Fixes #2150), and the #2271 responsive-settling work added regression tests for the controlled echo pattern.

**What we still need from you (@natle01):**

- Does this reproduce on v2.2.4 with the same 5.5% colWidth + margin config?
- If you use the controlled pattern (`layouts` + `onLayoutChange` echo), does the item still visibly move, or only fire `onLayoutChange` spuriously?
- Is `margin` passed as a per-breakpoint object or a flat tuple?
